src/models/model.cpp

```diff
@@ -277,167 +287,194 @@ void Model::InitDeviceAllocator([[maybe_unused]] OrtSession& session) {
 }
 
 void Model::CreateSessionOptions() {
-  session_options_ = OrtSessionOptions::Create();
-  auto& ort_options = *session_options_;
-  auto& options = config_->model.decoder.session_options;
-
-  // Default to a limit of 16 threads to optimize performance
-  constexpr int min_thread_nums = 1;
-  constexpr int max_thread_nums = 16;
-  int num_of_cores = std::max(min_thread_nums, static_cast<int>(std::thread::hardware_concurrency() / 2));
-  ort_options.SetIntraOpNumThreads(std::min(num_of_cores, max_thread_nums));
-
-  if (options.intra_op_num_threads.has_value()) {
-    ort_options.SetIntraOpNumThreads(options.intra_op_num_threads.value());
-  }
-
-  if (options.inter_op_num_threads.has_value()) {
-    ort_options.SetInterOpNumThreads(options.inter_op_num_threads.value());
-  }
-
-  if (options.enable_cpu_mem_arena.has_value()) {
-    if (options.enable_cpu_mem_arena.value())
-      ort_options.EnableCpuMemArena();
-    else
-      ort_options.DisableCpuMemArena();
-  }
-
-  if (options.enable_mem_pattern.has_value()) {
-    if (options.enable_cpu_mem_arena.value())
-      ort_options.EnableMemPattern();
-    else
-      ort_options.DisableMemPattern();
-  }
+  const auto create_session_options_from_config = [&](const Config::SessionOptions& config_session_options,
```

Comment:
Can this just be a separate function vs a lambda?

Reply:
I try to avoid changing the header file, if possible, to save on compile time. But I think it just makes reading better if this were a function of its own. Will make the change.